ophyd_devices/interfaces/base_classes/psi_positioner_base.py

```diff
@@ -0,0 +1,322 @@
+                signal.__init__(
+                    prefix + pv_suffix,
+                    name=self.name + self._child_name_separator + signal_name,
+                    **component.kwargs,
+                )
+
+        self.user_readback.subscribe(self._pos_changed)
+        self.motor_done_move.subscribe(self._move_changed)
+
+        # Make the default alias for the user_readback the name of the motor itself
+        # for compatibility with EpicsMotor
+        self.user_readback.name = self.name
+
+    def _remaining_defaults(self, attr: _SignalSentinel) -> set[str]:
+        return set(filter(lambda s: getattr(self, s) is attr, self.SIGNAL_NAMES))
+
+    @property
+    def egu(self):
+        """The engineering units (EGU) for a position"""
+        return self._egu
+
+    def check_value(self, pos):
+        """Check that the position is within the soft limits"""
+        if self.limits is not None:
+            low, high = self.limits
+            if low != high and not (low <= pos <= high):
+                raise ValueError("{} outside of user-specified limits" "".format(pos))
+        else:
+            self.user_setpoint.check_value(pos)
+
+    @property
+    def moving(self):
+        """Whether or not the motor is moving
+
+        If a `done` PV is specified, it will be read directly to get the motion
+        status. If not, it determined from the internal state of PVPositioner.
+
+        Returns
+        -------
+        bool
+        """
+        dval = self.motor_done_move.get(use_monitor=False)
+        return dval != self.done_value
+
+    def _setup_move(self, position):
+        """Move and do not wait until motion is complete (asynchronous)"""
+        self.log.debug("%s.user_setpoint = %s", self.name, position)
+        self.user_setpoint.put(position, wait=True)
+
+    def move(self, position, wait=True, timeout=None, moved_cb=None):
+        """Move to a specified position, optionally waiting for motion to
+        complete.
+
+        Parameters
+        ----------
+        position
+            Position to move to
+        moved_cb : callable
+            Call this callback when movement has finished. This callback must
+            accept one keyword argument: 'obj' which will be set to this
+            positioner instance.
+        timeout : float, optional
+            Maximum time to wait for the motion. If None, the default timeout
+            for this positioner is used.
+
+        Returns
+        -------
+        status : MoveStatus
+
+        Raises
+        ------
+        TimeoutError
+            When motion takes longer than `timeout`
+        ValueError
+            On invalid positions
+        RuntimeError
+            If motion fails other than timing out
+        """
+        # Before moving, ensure we can stop (if a stop_signal is configured).
+        if self.motor_stop is not _OPTIONAL_SIGNAL:
+            self.motor_stop.wait_for_connection()
```

Comment:
In BEC, we are calling wait_for_connection when we create the object on the device server. I guess that should be enough... or is there a specific reason to wait here?